fed/mosaic-application/src/main/java/org/eclipse/mosaic/fed/application/app/api/os/ElectricVehicleOperatingSystem.java

```diff
@@ -45,4 +46,11 @@ public interface ElectricVehicleOperatingSystem extends VehicleOperatingSystem {
      * Sends a request to stop charging the battery of the vehicle.
      */
     void sendChargingStopRequest();
+
+    /**
+     * Sends a request to locate all charging stations in the provided area.
+     *
+     * @param searchArea The area where the charging stations are search
+     */
+    void sendChargingStationDiscoveryRequest(GeoCircle searchArea);
```

Comment:
We should probably define the search area as a `GeoArea`.
```suggestion
    void sendChargingStationDiscoveryRequest(GeoArea searchArea);
```

Furthermore, we should also discuss what signatures are relevant.
I can think of:
- `getClosestChargingStation(GeoPosition position)` & `getClosestChargingStation()` -> returns the closest charging station either to the current position of the vehicle or to a specified position. E.g., when running low on battery.
    - Here we should also consider if we just want to consider spatial proximity or also the travel time. As charging stations at the opposite direction of a highway can actually be quite far -> `getShortestPathChargingStation(...)`
- `getChargingStationsInArea(GeoArea geoArea)` -> your use case
- `getChargingStationsWithinRange(double distance)` -> Get all charging stations that can be reached within a distance
- `getChargingStationsAlongRoute(double maxDetour)` -> Get all charging stations that are located along the route. This requires advanced routing though.

Reply:
We don't need these features for this task. I'd like to focus on the MVP tasks first and implement the remaining features when needed. 

Also, searching in a circle is very straightforward, unlike searching in a polygon. As far as my understanding of the `SpatialTreeTraverser` class goes, we would need to implement this first. Currently, we only support finding the nearest point, not all points within a given polygon (which is very counterintuitive to me. Why is there `SpatialTreeTraverser.InGeoPolygon` when it doesn't search in a polygon? @kschrab)

Reply:
Update: `SpatialTreeTraverser.InGeoPolygon` searches the polygon that contains a point, is that correct? 

Maybe we can add a bit of documentation here. I would have guessed that the `SpatialTreeTraverser.InPolygon` does the same thing as `SpatialTreeTraverser.InCircle` but for a polygon.

Reply:
Doing the search with circles is much easier computational wise. Therefore, I would first see if this implementation with circles is working and is sufficient for our use-cases.